### PR TITLE
Adds console link instance for Hub

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
This patch adds a ConsoleLink CR for Hub where
it is created in the post reconciler by adding
the Hub UI Route

This is shown up on the console as an icon for
easy user navigation

Signed-off-by: Puneet Punamiya <ppunamiy@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

File: pkg/reconciler/openshift/tektonhub/extension.go [hunk 1]
```diff
@@ -168,9 +173,165 @@ func (oe openshiftExtension) PreReconcile(ctx context.Context, tc v1alpha1.Tekto
...
+		cl.Spec.Href = baseURL
+
+		unstrObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(cl)
+		if err != nil {
+			return err
+		}
+		u.SetUnstructuredContent(unstrObj)
+		return nil
+	}
+}
```
Comment (PuneetPunamiya) [inline pkg/reconciler/openshift/tektonhub/extension.go:333]: Should we move this functions to a separate file

Comment (PuneetPunamiya) [reply pkg/reconciler/openshift/tektonhub/extension.go:333]: cc @nikhil-thomas  @sm43

File: pkg/reconciler/openshift/tektonhub/extension.go [hunk 2]
```diff
@@ -168,9 +173,165 @@ func (oe openshiftExtension) PreReconcile(ctx context.Context, tc v1alpha1.Tekto
...
+		hubConsoleLinkManifest := oe.manifest.Append()
+		if err := applyHubConsoleLinkManifest(&hubConsoleLinkManifest); err != nil {
+			return err
+		}
+
+		if err := consoleLinkTransform(ctx, &hubConsoleLinkManifest, th.Status.GetUiRoute()); err != nil {
+			return err
+		}
+
+		if err := createInstallerSet(ctx, oe.operatorClientSet, th, hubConsoleLinkManifest,
```
Comment (PuneetPunamiya) [inline pkg/reconciler/openshift/tektonhub/extension.go:203]: If the console link instance is not created should we mark the status of post reconciler with an error

Comment (nikhil-thomas) [reply pkg/reconciler/openshift/tektonhub/extension.go:203]: if the installerset error has any error we should return error and the logic outside the postReconcile call in the main reconciler should be  handling that error

**Review by vdemeester — APPROVED** [review]
We need to get this in 0.55.x release branch as well
/cc @nikhil-thomas

File: pkg/reconciler/openshift/tektonhub/extension.go [hunk 3]
```diff
@@ -45,9 +48,11 @@ import (
 )
 
 const (
-	hubprefix               string = "tekton-hub"
-	tektonHubAPIResourceKey string = "api"
-	tektonHubUiResourceKey  string = "ui"
+	hubprefix                  string = "tekton-hub"
+	tektonHubAPIResourceKey    string = "api"
+	tektonHubUiResourceKey     string = "ui"
+	CreatedByValue             string = "Tekton-Hub"
```
Comment (sm43) [inline pkg/reconciler/openshift/tektonhub/extension.go:54]:
```suggestion
	CreatedByValue             string = "TektonHub"
```
keeping similar to other component

**Comment by sm43** [comment]
/approve
LGTM

**Comment by nikhil-thomas** [comment]
/hold
until we finalize the logo and Link Section text

**Pull request merged**